src/hotspot/share/gc/parallel/psOldGen.cpp

```diff
@@ -381,6 +382,7 @@ void PSOldGen::post_resize() {
                       &ParallelScavengeHeap::heap()->workers() : NULL;
 
   // ALWAYS do this last!!
+  OrderAccess::storestore();
```

Comment:
Maybe update the comment to use less caps and '!'. Instead tie back to the function comment explaining that the barrier is needed to guarantee the order in which the data structures get visible to other threads.

Reply:
Good idea.  Here's the revised comment:

```
-  // ALWAYS do this last!!
+  // Ensure the space bounds are updated are made visible to other
+  // threads after the other data structures have been resized.
   OrderAccess::storestore();
```

Reply:
The second "are" should be an "and", right? Otherwise looks great!

Reply:
+1

Reply:
Drat.  Will fix.